fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/PushDownSumThroughJoinOneSide.java

```diff
@@ -0,0 +1,209 @@
+            return null;
+        }
+
+        Set<Slot> leftGroupBy = new HashSet<>();
+        Set<Slot> rightGroupBy = new HashSet<>();
+        for (Expression e : agg.getGroupByExpressions()) {
+            Slot slot = (Slot) e;
+            if (leftOutput.contains(slot)) {
+                leftGroupBy.add(slot);
+            } else if (rightOutput.contains(slot)) {
+                rightGroupBy.add(slot);
+            } else {
+                return null;
+            }
+        }
+        join.getHashJoinConjuncts().forEach(e -> e.getInputSlots().forEach(slot -> {
+            if (leftOutput.contains(slot)) {
+                leftGroupBy.add(slot);
+            } else if (rightOutput.contains(slot)) {
+                rightGroupBy.add(slot);
+            } else {
+                throw new IllegalStateException("Slot " + slot + " not found in join output");
+            }
+        }));
+
+        Plan left = join.left();
+        Plan right = join.right();
+
+        Map<Slot, NamedExpression> leftSumSlotToOutput = new HashMap<>();
+        Map<Slot, NamedExpression> rightSumSlotToOutput = new HashMap<>();
+
+        // left Sum agg
+        if (!leftSums.isEmpty()) {
+            Builder<NamedExpression> leftSumAggOutputBuilder = ImmutableList.<NamedExpression>builder()
+                    .addAll(leftGroupBy);
+            leftSums.forEach(func -> {
+                Alias alias = func.alias(func.getName());
+                leftSumSlotToOutput.put((Slot) func.child(0), alias);
+                leftSumAggOutputBuilder.add(alias);
+            });
+            left = new LogicalAggregate<>(ImmutableList.copyOf(leftGroupBy), leftSumAggOutputBuilder.build(),
+                    join.left());
+        }
+
+        // right Sum agg
+        if (!rightSums.isEmpty()) {
+            Builder<NamedExpression> rightSumAggOutputBuilder = ImmutableList.<NamedExpression>builder()
+                    .addAll(rightGroupBy);
+            rightSums.forEach(func -> {
+                Alias alias = func.alias(func.getName());
+                rightSumSlotToOutput.put((Slot) func.child(0), alias);
+                rightSumAggOutputBuilder.add(alias);
+            });
+            right = new LogicalAggregate<>(ImmutableList.copyOf(rightGroupBy), rightSumAggOutputBuilder.build(),
+                    join.right());
+        }
+
+        Preconditions.checkState(left != join.left() || right != join.right());
+        Plan newJoin = join.withChildren(left, right);
+
+        // top Sum agg
+        // replace sum(x) -> sum(sum#)
+        List<NamedExpression> newOutputExprs = new ArrayList<>();
+        for (NamedExpression ne : agg.getOutputExpressions()) {
+            if (ne instanceof Alias && ((Alias) ne).child() instanceof Sum) {
+                Sum oldTopSum = (Sum) ((Alias) ne).child();
+
+                Slot slot = (Slot) oldTopSum.child(0);
+                if (leftSumSlotToOutput.containsKey(slot)) {
```

Comment:
some as the above comments, do the definite side push transformation where without 'if-else' by moving some pre-processing at former logic?